src/tincctl.c

```diff
@@ -3384,6 +3264,7 @@ static int cmd_shell(int argc, char *argv[]) {
 }
 
 static void cleanup() {
+	free(cmdline_conf);
```

Comment:
The reason here is the same as in the sibling comment thread — to unbreak `read_config_options`. We can either add an additiional check here

https://github.com/gsliepen/tinc/blob/858370fad109168a670b521e47d4536bd2586102/src/conf.c#L361-L363

, or allocate `cmdline_conf`.

Thanks for the review, I'll fix everything later in the day.

Reply:
Ah, I see. I think adding the NULL-pointer check in `read_config_options()` is nicer, and not ad-hoc at all. Alternatively, we could make `cmdline_conf` a `list_t` instead of a `list_t *` in `conf.c`:

```
list_t cmdline_conf = {
    .delete = free_config,
};
```

But maybe that's a separate issue (I created #294 for that).

Reply:
This PR is already on the heavier side IMHO. I can pick this up a bit later after doing the belated types & compiler warnings cleanups.

Everything here should be fixed. Tests with e050753d1fc4d4465032919746c2a4db78f19932 reverted:
- https://github.com/hg/tinc/actions/runs/1078774256
- https://github.com/hg/tinc/actions/runs/1078774063
- https://github.com/hg/tinc/actions/runs/1078773738